### PR TITLE
Pushing an existing version of a gem should not overwrite the gem con…

### PR DESCRIPTION
**Pull request by bronzdoc**
closes https://github.com/bundler/gemstash/issues/188

### CODE REVIEW

**Comment by bronzdoc** [comment]
A spec in jruby is failing but i can't reproduce the error locally...

Running `bundle rake rspec`
```shell
.....
Finished in 1 minute 46.58 seconds (files took 6.38 seconds to load)
257 examples, 0 failures, 3 pending
```

```shell
$ rbenv versions
  system
  2.2.10
  2.3.7
  2.4.1
  2.4.4
  2.5.0
  2.5.1
* jruby-9.2.0.0 (set by /home/bronzdoc/projects/ruby/gemstash/.ruby-version)
```

The build fails in ruby-2.4.1 and ruby and in ruby-2.3.4 got fixed by restarting the builds.

**Comment by bbuchalter** [comment]
> An integration spec in jruby is failing but i can't reproduce the error locally...
> The build failing in ruby-2.4.1 and ruby and in ruby-2.3.4 got fixed by restarting the builds.

https://travis-ci.org/bundler/gemstash/jobs/440896430#L513: `The job exceeded the maximum time limit for jobs, and has been terminated.`

Maybe we can retry the jruby build too?

**Comment by bronzdoc** [comment]
@bbuchalter 
```
The job exceeded the maximum time limit for jobs, and has been terminated.
```

^ That is me running the build in `debug` mode, travis allows you to get inside the build container to debug, that is not the actual failure.

I will work on it later tonight

File: lib/gemstash/gem_pusher.rb [hunk 1]
```diff
@@ -53,6 +53,13 @@ def check_auth
     end
 
     def store_gem
+      resource_exist = storage.resource(full_name).exist?
+      resource_is_indexed = storage.resource(full_name).properties[:indexed] if resource_exist
+
+      if resource_exist && resource_is_indexed
```
Comment (bbuchalter) [inline lib/gemstash/gem_pusher.rb]: This guard is mirrored in [`save_to_database`](https://github.com/bundler/gemstash/blob/f14894610a1cbef21d6f595ba3b332e45baa5f32/lib/gemstash/gem_pusher.rb#L79). Perhaps it should be extracted to a private method?

Also, there is a [`YankedVersionError` guard](https://github.com/bundler/gemstash/blob/f14894610a1cbef21d6f595ba3b332e45baa5f32/lib/gemstash/gem_pusher.rb#L80) in `save_to_database`. Should that _also_ be here?

**Comment by bronzdoc** [comment]
For the record... the build failure is not related to this PR, the failure is happening in `master`
 
The specs pass if change the ruby version to `jruby-9.1.17.0` inside the build container.

File: lib/gemstash/gem_pusher.rb (hunk 1 above)
Comment (bronzdoc) [reply lib/gemstash/gem_pusher.rb]: WIll do a followup PR if applies, thanks

**Comment by bronzdoc** [comment]
@bundlerbot r+

**Pull request merged**